go/client/logclient.go

```diff
@@ -29,8 +29,18 @@ const (
 	GetEntriesPath  = "/ct/v1/get-entries"
 )
 
-// LogClient represents a client for a given CT Log instance
-type LogClient struct {
+// LogClient defines an interface for Log Clients.  This aids testing.
+type LogClient interface {
+	AddChain(chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
+	AddPreChain(chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
+	AddChainWithContext(ctx context.Context, chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error)
+	AddJSON(data interface{}) (*ct.SignedCertificateTimestamp, error)
+	GetSTH() (sth *ct.SignedTreeHead, err error)
+	GetEntries(start, end int64) ([]ct.LogEntry, error)
+}
+
+// logClientImpl conforms to LogClient and represents a client for a given CT Log instance
+type logClientImpl struct {
```

Comment:
The Effective Go sections on naming don't really go into this btw.  A quick strawpoll of folk nearby suggests that having logClient and LogClient in one file is asking for trouble.